veps/sig-compute/secure-execution.md

```diff
@@ -0,0 +1,64 @@
+# Overview
+Implement support for IBM Z Secure Execution VMs in kubevirt
+
+## Motivation
+To enable customers to deploy secure workloads on their IBM Z machines using kubevirt.
+
+## Goals
+To enable kubevirt to run Secure Execution VMs on s390x
+
+## Non Goals
+Changing how existing VMs run.
+
+## Definition of Users
+- IBM Z users wanting to secure their new VMs using Secure Execution
+- IBM Z users wanting to migrate existing Secure Execution VMs to kubevirt
+
+## User Stories
+(list of user stories this design aims to solve)
+
+## Repos
+https://github.com/kubevirt/kubevirt
+https://github.com/kubevirt/api
+
+# Design
+For implementing this feature, the api would need to be expanded to have a Secure Execution option under Launch Security next to AMD SEV.
+Similiar to SEV it would be first behind a Feature Gate.
+
+Afterwards kubevirt would need to be expanded to handle setting the correct Launch Security type for Secure Execution.
+Lastly virt-controller would need to be modified to support starting virt-launcher pods with the correct capabilities and devices needed for Secure Execution.
+
+## API Examples
+```
+apiVersion: kubevirt.io/v1
+kind: VirtualMachineInstance
+metadata:
+  ...
+spec:
+  domain:
+    launchSecurity:
+      secureExecution: {}
```

Comment:
I'm worried about an accumulation of CPU/Arch specific fields here.
I wonder if instead of using multiple dedicated fields we could simplify it by using a generic term, such as `secureExecution`. 
Then we could have a generic "SecureExecution" device plugin that would have value on any node that has support for either one of the technologies. Once the vmipod is scheduled to the node it would pass the correct device. For migrations, it could work the same way as it works for host-model cpu.

@alicefr what do you think?

Reply:
I would add that the generic should if we choose to do it be `launchSecurity`, since that is what libvirt is using.
Also while i don't know about AMD SEV, but IBM Z Secure Execution is a CPU feature, so while it needs special permissions in the virt-launcher Pod, it does not really need a device plugin.

Reply:
> 2\. I wonder how different the parameters are for the different technologies. IOW how different the configs for AMV SEV, Intel TDX and IBM secure execution are.

I would naively just map it 1:1 to what libvirt uses in the [xml](https://libvirt.org/formatdomain.html#launch-security). That way the differences should not matter.  I mean IBM Secure Execution does not use any of the parameters except type anyway to my knowledge.

Reply:
We could maybe do something more generic by using maps:
```
type LaunchSecurity struct {
    Type string
    Properties map[string]string
}
```

Reply:
@alicefr Yes, I understand your point. It concerned me that we will have multiple of these structs. The number will obviously grow in the future. In many cases, the users won't care specifically about the technology that is being used to achieve this functionality. 
On the other hand, we need to increase the number of validations we do, to ensure that various combinations are sensible.
I was hoping we could somehow generalize the API so we could have some kind of an envelope for a generic API and it could remain empty for users who don't care about specific configuration but also can be expanded for those who care. 
A pointer to an external CRD is an interesting idea.

Reply:
I'm marking this as resolved as we have a separate mail thread now and i managed to drop the requirement for api changes.